src/localization/gps_linearization.py

```diff
@@ -28,59 +31,90 @@ class GPSLinearization:
     ref_lon: float
     ref_alt: float
 
-    # frame configuration
+    # frame/publishing configuration
+    publish_tf: bool
     use_odom: bool
     world_frame: str
     odom_frame: str
     rover_frame: str
 
-    def __init__(self):
-        # init to zero pose
-        self.pose = SE3()
-
-        rospy.Subscriber("gps/fix", NavSatFix, self.gps_callback)
-        rospy.Subscriber("imu/data", ImuAndMag, self.imu_callback)
-
-        self.tf_broadcaster = tf2_ros.TransformBroadcaster()
-        self.tf_buffer = tf2_ros.Buffer()
-        self.tf_listener = tf2_ros.TransformListener(self.tf_buffer)
+    # covariance config
+    use_dop_cov: bool
+    config_gps_covariance: np.ndarray
 
+    def __init__(self):
         # read required parameters, if they don't exist an error will be thrown
         self.ref_lat = rospy.get_param("gps_linearization/reference_point_latitude")
         self.ref_lon = rospy.get_param("gps_linearization/reference_point_longitude")
         self.ref_alt = rospy.get_param("gps_linearization/reference_point_altitude")
 
-        self.use_odom = rospy.get_param("gps_linearization/use_odom_frame")
-        self.world_frame = rospy.get_param("gps_linearization/world_frame")
-        self.odom_frame = rospy.get_param("gps_linearization/odom_frame")
-        self.rover_frame = rospy.get_param("gps_linearization/rover_frame")
+        self.publish_tf = rospy.get_param("gps_linearization/publish_tf")
+        self.use_odom = rospy.get_param("use_odom_frame")
+
+        self.world_frame = rospy.get_param("world_frame")
+        self.odom_frame = rospy.get_param("odom_frame")
+        self.rover_frame = rospy.get_param("rover_frame")
+
+        self.use_dop_cov = rospy.get_param("global_ekf/use_gps_dop_covariance")
```

Comment:
what is "dop_cov"?

Reply:
dilution of precision (DoP) covariance. it's explained in localization.yaml but I can add another comment here if you think its necessary. Basically the GPS driver will estimate covariance through [dilution of precision](https://en.wikipedia.org/wiki/Dilution_of_precision_(navigation)) and publish that covariance matrix as the GPS covariance matrix, so this option is whether to keep that or to overwrite it with the configured one.

Reply:
cool

Reply:
definitely would add a hefty comment here basically saying what u just said to me

Reply:
you don't think that should go in the wiki instead?

Reply:
probably should go there too. I'm just trying to get the code itself into a more readable state. Right now in general the code is just very hard to read and follow.  It took me nearly an hour to get through this review and I have a pretty good understanding of the loc architecture to begin with. That means that just explaining things in the wiki isn't going to be enough. The more inline explanations we can have in this the better.

Reply:
to be clear, not saying its hard to follow because its bad code. I'm saying its hard to follow mostly because loc stuff is just confusing and theres a couple layers of indirection.